tests/test_finder.py

```diff
@@ -1,65 +1,164 @@
-    test_file.write_bytes(b"!<        ustar")
-    results = search_yara_patterns(rules, handler_map, test_file)
+    def calculate_chunk(self, file, start_offset: int):
+        raise EOFError()
 
-    assert len(results) == 2
-    result1, result2 = results
 
-    assert result1.handler is handler1
-    assert result1.match.strings == [(0, "$handler1_magic", b"!<")]
+class TestHandlerInvalid(Handler):
+    NAME = "handlerEOF"
+    PATTERNS = [Regex("I")]
 
-    assert result2.handler is handler2
-    assert result2.match.strings == [(10, "$tar_magic", b"ustar")]
+    def calculate_chunk(self, file, start_offset: int):
+        raise InvalidInputFormat()
 
 
-def test_make_handler_map():
-    handler_map = make_handler_map(tuple([TestHandler1, TestHandler2]))
-    assert isinstance(handler_map["handler1"], TestHandler1)
-    assert isinstance(handler_map["handler2"], TestHandler2)
+class TestHandlerExc(Handler):
+    NAME = "handlerEXC"
+    PATTERNS = [Regex("EXC")]
 
+    def calculate_chunk(self, file, start_offset: int):
+        raise Exception("Error")
 
-def test_make_handler_map_instances_are_cached():
-    handler_map1 = make_handler_map(tuple([TestHandler1, TestHandler2]))
-    handler_map2 = make_handler_map(tuple([TestHandler1, TestHandler2]))
-    assert handler_map1["handler1"] is handler_map2["handler1"]
-    assert handler_map1["handler2"] is handler_map2["handler2"]
+
+def test_build_hyperscan_database():
+    db, handler_map = build_hyperscan_database((TestHandlerA, TestHandlerB))
+    matches = []
+    db.scan(
+        [bytearray(b"A123456789BB")],
+        match_event_handler=lambda pattern_id, start, end, flags, m: m.append(
+            (pattern_id, start, end)
+        ),
+        context=matches,
+    )
+
+    assert len(handler_map) == 3
+
+    assert len(matches) == 2
+    assert isinstance(handler_map[matches[0][0]], TestHandlerA)
+    assert isinstance(handler_map[matches[1][0]], TestHandlerB)
+    assert matches[0][1] == 0
+    assert matches[1][1] == 10
+
+
+def test_db_and_handler_map_instances_are_cached():
+    db1, handler_map1 = build_hyperscan_database(tuple([TestHandlerA, TestHandlerB]))
+    db2, handler_map2 = build_hyperscan_database(tuple([TestHandlerA, TestHandlerB]))
+    db3, handler_map3 = build_hyperscan_database(
+        tuple(
+            [
+                TestHandlerA,
+            ]
+        )
+    )
+    assert db1 is db2
+    assert handler_map1 is handler_map2
+    assert db1 is not db3
+    assert handler_map1 is not handler_map3
+
+
+def test_invalid_hexstring_pattern_raises():
+    class InvalidHandler(Handler):
+        NAME = "InvalidHandler"
+        PATTERNS = [Hexstring("invalid pattern")]
+
+        def calculate_chunk(self, file, start_offset: int):
+            pass
+
+    with pytest.raises(ValueError):
+        build_hyperscan_database(tuple([TestHandlerA, TestHandlerB, InvalidHandler]))
+
+
+@pytest.mark.parametrize(
+    "content, expected_chunks",
```

Comment:
The `expected_chunks` in these parameters are in-fact, just tuples with numbers. Can't they be `Chunk`s (or `ValidChunk`?)?

Reply:
done